lib/pavilion/test_config/file_format.py

```diff
@@ -121,25 +121,38 @@ class EnvCatElem(yc.CategoryElem):
     type = OrderedDict
 
 
-class TestConfigLoader(yc.YamlConfigLoader):
-    """This class describes a test section in a Pavilion config file. It is
-expected to be added to by various plugins.
-
-:cvar list(yc.YamlConfig) ELEMENTS: Each YamlConfig instance in this
-    list defines a key for the test config.
-
-- Each element must result in a string (which is why you see a lot of StrElem
-  below), or a structure that contains only strings at the lowest layer.
+class SeriesConfigLoader(yc.YamlConfigLoader):
+    """This class describes a series file."""
 
-  - So lists of dicts of strings are fine, etc.
-  - yc.RegexElem also produces a string.
-- Everything should have a sensible default.
+    ELEMENTS = [
+        yc.CategoryElem(
+            'series', sub_elem=yc.KeyedElem(
+                elements=[
+                    yc.ListElem('depends_on', sub_elem=yc.StrElem()),
+                    yc.ListElem('test_names', sub_elem=yc.StrElem()),
+                    yc.ListElem('modes', sub_elem=yc.StrElem()),
+                    yc.CategoryElem(
+                        'only_if', sub_elem=yc.ListElem(sub_elem=yc.StrElem())
+                    ),
+                    yc.CategoryElem(
+                        'not_if', sub_elem=yc.ListElem(sub_elem=yc.StrElem())
+                    )
+                ]
+            ),
+        ),
```

Comment:
So these all apply globally?
  - modes makes sense
  - Does 'on_complete' happen after 'all' series defined in this file, or 'each'.

Reply:
Shouldn't there be an 'on_complete' at the series level too?